### PR TITLE
Ablog is NOT safe for parallel read

### LINKED ISSUE #297: Ablog isn't safe for parallel reading
### Describe the bug

It looks Ablog isn't safe for parallel reading (introduced in ad5665a).

But it's safe for parallel writing (4f9ae64).

Both changes are recent (summer this year) and made by @nabobalis :-) What lead you to these changes? :-)

I've add `-j auto` to sphinx-build (use as many CPU cores as possible) recently and noticed Ablog doesn't detect any single post and all `postlists` were empty. With `-j 1` (use single core, default value) Ablog works again.

It would be nice to fix it, but I am unsure about the reasons. As "workaround" I suggest change to `"parallel_read_safe": False, "parallel_write_safe": True`.

### To Reproduce

_No response_

### Screenshots

_No response_

### System Details

_No response_

### Installation method

_No response_

### PR DESCRIPTION
**Pull request by liborjelinek**
## PR Description

fixes #297

### CODE REVIEW

File: src/ablog/tests/test_parallel.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    init_py_path.write_text(bad_init_py)
+
+    # liborjelinek: I wasn't able to demonstrate the issue with the `parallel` argument to the `sphinx` fixture
+    # @pytest.mark.sphinx("html", testroot="parallel", parallel=2)
+    # therefore running sphinx-build externally
+    indir = rootdir / "test-parallel"
+    run(["sphinx-build", "-M", "html", indir.as_posix(), "_build", "-j", "auto"], check=True)
+
+    outdir = indir / "_build" / "html"
+    assert (outdir / "postlist.html").exists()
```
Comment (nabobalis) [inline src/ablog/tests/test_parallel.py:23]:
```
=========================== short test summary info ============================
FAILED .tox/py312-sphinx8/lib/python3.12/site-packages/ablog/tests/test_parallel.py::test_not_safe_for_parallel_read - AssertionError: assert False
 +  where False = exists()
 +    where exists = (PosixPath('/home/runner/work/ablog/ablog/roots/test-parallel/_build/html') / 'postlist.html').exists
```
This file doesn't seem to exist when run on github actions.

Comment (liborjelinek) [reply src/ablog/tests/test_parallel.py:23]: You're right. I have tested it many times locally. It looks that GitHub Actions runner have some security limits to read files created... I'll try to consult the docs.

Comment (liborjelinek) [reply src/ablog/tests/test_parallel.py]: A note to myself: Yes, GH Actions really somehow prevent to access created files. I tried `sudo ls -l` of a file produced by test and it
```
ls: cannot access '/home/runner/work/ablog/ablog/roots/test-parallel/_build/html/postlists.html': No such file or directory
```

Comment (nabobalis) [reply src/ablog/tests/test_parallel.py]: I wonder if it's possible to use pytest's tmpdir fixture to write the output ablog files there?

Comment (liborjelinek) [reply src/ablog/tests/test_parallel.py]: Okay, it was a the stupidest error ever :-) The created file is `postlist.html`, not `postlists.html`. However, I've kept building into Pytest temp dir. Good idea anyway. Test is passing now.

Comment (nabobalis) [reply src/ablog/tests/test_parallel.py]: Thanks!

**Review by nabobalis — APPROVED** [review]

**Pull request merged**